CONTRIBUTING.md

```diff
@@ -0,0 +1,217 @@
+   * [GitHub](https://github.com/ballerina-platform/ballerina-lang/issues): file issues, comment on other issues, send your pull requests. 
+* Submit issues:
+   * Found a security flaw? Please email security@ballerina.io
+   * Submitting a bug is just as important as contributing code. Go to the Issues tab of the GitHub repo and click the New Issue button to file a bug report.
+* Start with easy fix issues:
+   * Browse issues labeled easyfix,
+   * Use comments on the issue itself to indicate that you will be working on it and get guidance and help.
+
+## Communicating with the team
+
+[Ballerina-Dev Google Group](https://groups.google.com/forum/#!forum/ballerina-dev) is the main Ballerina project discussion forum.
+
+[StackOverflow](https://stackoverflow.com/questions/tagged/ballerina) is used for support, [Slack](https://ballerinalang.slack.com/) for real-time communications, and [GitHub](https://github.com/ballerina-platform/ballerina-lang/issues) for issues and code repositories.
+
+## Filing issues
+
+If you are unsure whether you have found a bug, please consider searching existing issues in github and asking in Ballerina-Dev Google Group.
+
+> IMPORTANT: Sensitive security-related issues should be reported to [security@ballerina.io](security@ballerina.io). See the security policy for details.
+
+To file non-security issues:
+
+1. Click the Issues tab in the github repository,
+
+2. Click the New Issue button,
+
+3. Fill out all sections in the issue template and submit.
+
+## Contributing site changes, docs, and examples
+
+## Contributing code
+
+### Accepting Contributor License Agreement (CLA)
+
+Before you submit your first contribution please accept our Contributor License Agreement (CLA) here. When you send your first Pull Request (PR), GitHub will ask you to accept the CLA.
+
+There is no need to do this before you send your first PR.
+
+Subsequent PRs will not require CLA acceptance.
+
+If for some (unlikely) reason at any time CLA changes, you will get presented with the new CLA text on your first PR after the change.
+
+### Obtaining the Source Code and Building the Project
+
+#### Build Prerequisites
+
+**Prerequisite 1** - Install a Java Development Kit (JDK) version 8
+
+Building Ballerina requires a Java SE Development Kit (JDK) version 8 to be installed. You can download one from one of the following locations:
+
+* [http://www.oracle.com/technetwork/java/javase/downloads/index.html](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [http://openjdk.java.net/install/index.html](http://openjdk.java.net/install/index.html)
+
+> NOTE: Set an environment variable `JAVA_HOME` to the path name of the directory into which you installed the JDK release.
+
+**Prerequisite 2** - Install Maven 3.3+ or later 
+
+Install maven 3.3 or a later version: [https://maven.apache.org/install.html](https://maven.apache.org/install.html)

```

Comment:
The build doesn't work on Maven 3.3. The dev team switched to 3.5.3, which is the latest.